### PR TITLE
Simplify HD paths

### LINKED ISSUE #51: User simpler HD paths
In https://github.com/iov-one/iov-faucet/blob/v0.9.0/README.md#faucet-hd-wallet, we define a faucet specific way to derive HD paths. The reason for that is that we wanted the ability to spawn different faucet processes using the same mnemonic (see `instance_index`). However, we never spawned more than one so this custom HD schema can easily replaced with the standard account paths that we use in the wallet.

A function very similar to

```ts
export function pathBuilderForCodec(codecType: CodecType): (derivation: number) => readonly Slip10RawIndex[] {
  const pathBuilder = (derivation: number): readonly Slip10RawIndex[] => {
    switch (codecType) {
      case CodecType.Bns:
        return HdPaths.iov(derivation);
      case CodecType.Lisk:
        return HdPaths.bip44Like(134, derivation);
      case CodecType.Ethereum:
        return HdPaths.ethereum(derivation);
      default:
        throw new Error(`unsupported codec: ${codecType}`);
    }
  };
  return pathBuilder;
}
```

could be used to expose the coin specific paths via a generic interface.

If you really need multiple instances, you can use different mnemonics.

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,21 +130,22 @@ curl --header "Content-Type: application/json" \
...
-| ---------- | -------- | --------------- | ---------------- |
-| 0.10.x     | 1.2.x    | 0.22.x          |                  |
-| 0.9.x      | 0.17.x   | 0.21.x          |                  |
-| 0.8.x      | 0.16.x   | 0.19.x – 0.20.x |                  |
-| 0.7.x      | 0.15.x   | 0.16.x          |                  |
-| 0.6.x      | 0.14.x   | 0.14.x          | BNS fee support  |
-| 0.5.x      | 0.12.x   | 0.10.x – 0.11.x |                  |
-| 0.4.x      | 0.11.x   | 0.4.x – 0.9.x   | Ethereum support |
-| 0.3.x      | 0.9.x    | 0.4.x – 0.9.x   |                  |
+| iov-faucet | IOV-Core | BNSd support    | New features                                                              |
```
Comment (willclarktech) [inline README.md]: Was this table formatted by prettier?

File: src/hdpaths.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// tslint:disable:no-unused-expression
```
Comment (willclarktech) [inline src/hdpaths.spec.ts]: Why is this necessary? Do we not have the tslint plugin?

File: README.md (hunk 1 above)
Comment (webmaster128) [reply README.md]: Yes it was, by running `yarn format-text`

File: src/hdpaths.spec.ts (hunk 2 above)
Comment (webmaster128) [reply src/hdpaths.spec.ts]: Jap, no chai/tslint plugin installed:

```
    "@types/bn.js": "^4.11.4",
    "@types/chai": "^4.1.7",
    "@types/koa": "^2.0.48",
    "@types/mocha": "^5.2.5",
    "chai": "^4.2.0",
    "mocha": "^5.2.0",
    "prettier": "^1.18.2",
    "tslint": "^5.18.0",
    "tslint-config-prettier": "^1.18.0",
    "tslint-immutable": "^6",
    "typescript": "~3.7"
```

Let's do that for eslint instead of tslint: https://github.com/iov-one/iov-faucet/issues/54

File: README.md (hunk 1 above)
Comment (willclarktech) [reply README.md]: It doesn't look like the columns all have the same length for each row.

Comment (webmaster128) [reply README.md]: I little bit hard to review because of the long lines but they do: https://raw.githubusercontent.com/iov-one/iov-faucet/simplify-path/README.md
